core/headerchain.go

```diff
@@ -60,20 +60,25 @@ type HeaderChain struct {
 
 	headermu sync.RWMutex
 	heads    []*types.Header
+
+	getPendingEtxsFromSub GetPendingEtxsFromSub
 }
 
+type GetPendingEtxsFromSub func(common.Hash, common.Location) (types.PendingEtxs, error)
```

Comment:
Why doesn't this accept the `HashAndLocation` type you created above?

Reply:
Actually, I like the way this (and related) code is written. I wonder if we need the `HashAndLocation` type at all